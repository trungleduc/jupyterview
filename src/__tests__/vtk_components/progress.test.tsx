import ProgressNotification from "../../vtk_components/progress";
import { mount, shallow, render, ShallowWrapper } from "enzyme";

import React from "react";
import {mock_send_msg, MockModel} from "../../test_setup/utils"
describe("Test ProgressNotification component", () => {
  let component: ShallowWrapper
  beforeAll(() => {
    const  model = new MockModel()
    component = shallow(<ProgressNotification open = {true} value = {0} />);
  })

  it("should render correctly with mock props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should have correct initial state", () => {
    expect(component.state("autoFocus")).toEqual(false)
    expect(component.state("canEscapeKeyClear")).toEqual(true)
    expect(component.state("position")).toEqual("top-right")
  });

});
