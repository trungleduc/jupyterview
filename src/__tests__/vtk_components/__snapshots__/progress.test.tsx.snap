// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProgressNotification component should render correctly with mock props 1`] = `<div />`;
